tests/test_backend_general.py

```diff
@@ -0,0 +1,630 @@
+import logging
+
+import calliope
+import numpy as np
+import pandas as pd
+import pytest  # noqa: F401
+import xarray as xr
+
+from .common.util import build_test_model as build_model
+from .common.util import check_error_or_warning
+
+
+@pytest.fixture(scope="class", params=["pyomo", "gurobi"])
```

Comment:
shouldn't the scope of these be "module"? Is there an advantage to "class" in this case?

Reply:
Some classes end up changing the backend model and I'd prefer that not to bleed between classes